Support IPv6

melle opened the pull request.

melle committed:
Add support for IPv6 link local nodes (ff02::1%en0)

- UDPBroadcastConnection is now the base class, v4/v6 specific code was extracted to
  subclasses UDPv4BroadcastConnection / UDPv6BroadcastConnection
- BUG: bindIt == true is not yet supported in UDPv6BroadcastConnection

melle committed:
IPv6 support for the README as well

zweigraf commented on UDPBroadcast/UDPv6BroadcastConnection.swift:
```diff
@@ -0,0 +1,131 @@
+        // Bind socket if needed
+        if shouldBeBound {
+            // FIXME: not yet implemented
+            debugPrint("FIXME: implement IPv6 binding!")
```
Will this be done?

melle replied:
Probably not. This was added to the original project by a 3rd party but we will never use it. I'm tempted to remove this addition to keep the whole thing as simple as possible.

zweigraf commented:
If we use this in production, I would love to see some better types on the API of this thing. (IP address not as String, ports not as Int, etc).

melle committed:
Remove bindIt parameter (breaking compatibility with the upstream library)

- we have no need for this, it simplifies the lib

melle committed:
Code review feedback

- removed abstract base class, moved everything back into one class
- there is a bit of code duplication in favour of complex abstractions

melle commented:
ok, this breaks compatibility with upstream (removed bindIt, added addressFamily). Higher level types (like Network.framework's IPv6Address) should not be used here as this adds a dependency to iOS 12.

zweigraf commented on UDPBroadcast/UDPBroadcastConnection.swift:
```diff
@@ -150,11 +267,81 @@ open class UDPBroadcastConnection: UDPBroadcastSocketCreating {
+    ///
+    /// - Parameter data: Data to send via broadcast.
+    /// - Throws: Throws a `ConnectionError` if an error occurs.
+    private func sendBroadcastv4(data: Data) throws {
```
Aren't those two functions here basically the same, apart from `v4address` and `v6address`?

melle replied:
yes, I tried to untangle this yesterday with Luizb and failed. We can retry to reduce the redundancy later.
